### PR TITLE
Transaction filtering by time range

### PR DESCRIPTION
**Pull request by theDoubleBubble**

### CODE REVIEW

File: src/Core/Transaction/Transaction.actions.js [hunk 1]
```diff
@@ -68,8 +70,9 @@ export const fetchTransactionsForProject = (projectSlug, username, filters, page
                     contractId: contractsFilter ? contractsFilter.value.map(contractId => Contract.generateApiId(contractId)) : null,
                     network: networksFilter ? getApiIdForNetwork(networksFilter.value) : null,
                     tag: tagFilter ? tagFilter.value : null,
-                    after: afterFilter ? afterFilter.value : null,
-                },
+                    after: (afterFilter && afterFilter.value instanceof moment) ? afterFilter.value.utc().format() : null,
+                    before: (beforeFilter && beforeFilter.value instanceof moment) ? beforeFilter.value.utc().format() : null,
```
Comment (miljantekic) [inline src/Core/Transaction/Transaction.actions.js:74]: I think there is no need to check if the value is an instance of `moment`. We can assume that the value can be a `string` or `moment` object and just cast it either way into a moment object and then convert into utc. Moment doesn't have problems when constructed with a moment object

```
moment(beforefilter.value)
```

File: src/Components/DateRangePicker/DateRangePicker.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    />
+                </div>
+            </div>
+        )
+    }
+}
+
+DateRangePicker.propTypes = {
+    after: momentPropTypes.momentObj,
+    before: momentPropTypes.momentObj,
```
Comment (miljantekic) [inline src/Components/DateRangePicker/DateRangePicker.js:52]: I wouldn't force the `after` and `before` properties to be moment instances. We should probably support accepting both strings and moment objects

File: src/Components/TransactionFilters/TransactionFilters.js [hunk 3]
```diff
@@ -197,6 +205,34 @@ class TransactionFilters extends Component {
...
     };
 
+    handleBeforeChange = (value) => {
+        const { draftAfter } = this.state;
+
+        this.setState({
+            draftBefore: value
+        })
+
+        if ( draftAfter && value && draftAfter > value ) {
```
Comment (miljantekic) [inline src/Components/TransactionFilters/TransactionFilters.js:215]: Use moments `isAfter` and `isBefore` methods for checking this as to not rely on basic comparison operators.

File: src/Components/TransactionFilters/TransactionFilters.js [hunk 4]
```diff
@@ -197,6 +205,34 @@ class TransactionFilters extends Component {
...
+    }
+
+    handleAfterChange = (value) => {
+        const { draftBefore } = this.state;
+
+        this.setState({
+            draftAfter: value
+        })
+
+        if ( value && draftBefore && value > draftBefore ) {
```
Comment (miljantekic) [inline src/Components/TransactionFilters/TransactionFilters.js:231]: same as above comment for `isBefore` and `isAfter`

File: src/Components/TransactionFilters/TransactionFilters.js [hunk 5]
```diff
@@ -256,8 +300,13 @@ class TransactionFilters extends Component {
         });
 
         filters.push({
-            type: TransactionFilterTypes.AFTER,
-            value: draftAfter ? draftAfter : null,
+        type: TransactionFilterTypes.AFTER,
+        value: draftAfter ? draftAfter : null,
+        })
```
Comment (miljantekic) [inline src/Components/TransactionFilters/TransactionFilters.js:305]: Formatting and semicolon missing

File: package.json [hunk 6]
```diff
@@ -32,12 +32,14 @@
...
     "react-blockies": "^1.4.1",
     "react-copy-to-clipboard": "^5.0.1",
+    "react-dates": "^21.8.0",
     "react-dom": "^16.9.0",
     "react-ga": "^2.5.3",
     "react-google-login": "^4.0.0",
     "react-helmet": "^5.2.0",
     "react-json-view": "^1.19.1",
     "react-modal": "^3.5.1",
+    "react-moment-proptypes": "^1.7.0",
```
Comment (miljantekic) [inline package.json:42]: Potentially check if this proptypes library is actually needed

**Comment by miljantekic** [comment]
@theDoubleBubble There is also a conflict with the master branch so just do a merge of master and after you resolve the comments above we can put this on staging to be tested

File: package.json (hunk 6 above)
Comment (theDoubleBubble) [reply package.json:42]: This is used for propType in DateRangePicker.js for after and before

**Pull request closed without merging**